tests/data_context/test_data_context_datasource_runtime_data_connector.py

```diff
@@ -359,3 +1167,347 @@ def test_get_validator_wrong_runtime_parameters_sqlalchemy_engine(
+    assert batch.batch_spec is not None
+    assert batch.batch_definition["data_asset_name"] == "my_data_asset"
+    assert isinstance(batch, Batch)
+    assert isinstance(batch.data, SparkDFBatchData)
+    assert batch.data.dataframe.count() == 10000  # 3 files read in as 1
+    assert len(batch.data.dataframe.columns) == 18
+
+
+def test_get_batch_spark_directory_fail_no_reader_method(
+    data_context_with_datasource_spark_engine, taxi_test_file_directory, spark_session
+):
+    context = data_context_with_datasource_spark_engine
+    with pytest.raises(ge_exceptions.ExecutionEngineError):
+        batch_list: list = context.get_batch_list(
+            batch_request=RuntimeBatchRequest(
+                datasource_name="my_datasource",
+                data_connector_name="default_runtime_data_connector_name",
+                data_asset_name="default_data_asset_name",
+                runtime_parameters={
+                    "path": taxi_test_file_directory,
+                },
+                batch_identifiers={"default_identifier_name": 1234567890},
+            )
+        )
+
+
+def test_get_batch_spark_directory_fail_wrong_reader_method(
+    data_context_with_datasource_spark_engine, taxi_test_file_directory, spark_session
+):
+    context = data_context_with_datasource_spark_engine
+    with pytest.raises(ge_exceptions.ExecutionEngineError):
+        batch_list: list = context.get_batch_list(
+            batch_request=RuntimeBatchRequest(
+                datasource_name="my_datasource",
+                data_connector_name="default_runtime_data_connector_name",
+                data_asset_name="default_data_asset_name",
+                runtime_parameters={
+                    "path": taxi_test_file_directory,
+                },
+                batch_identifiers={"default_identifier_name": 1234567890},
+                batch_spec_passthrough={
+                    "reader_options": {"header": True},
+                    "reader_method": "i_am_invalid",
+                },
+            )
+        )
+
+
+def test_sparkdf_execution_engine_batch_definition_list_from_batch_request_success_file_path_no_header(
+    data_context_with_datasource_spark_engine,
+    taxi_test_file,
+    spark_session,
+):
+    context = data_context_with_datasource_spark_engine
+    batch_list: list = context.get_batch_list(
+        batch_request=RuntimeBatchRequest(
+            datasource_name="my_datasource",
+            data_connector_name="default_runtime_data_connector_name",
+            data_asset_name="my_data_asset",
+            runtime_parameters={
+                "path": taxi_test_file,
+            },
+            batch_identifiers={"default_identifier_name": 1234567890},
+            batch_spec_passthrough={
+                "reader_method": "csv",
+            },
+        )
+    )
+    assert len(batch_list) == 1
+    assert isinstance(batch_list[0], Batch)
+    batch = batch_list[0]
+    assert batch.batch_spec is not None
```

Comment:
For these batch spec assertions, do we maybe want to assert `isinstance`?

Reply:
really good catch. Thank you 🙇🏼